product_listing.go

```diff
@@ -0,0 +1,148 @@
+package goshopify
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const productListingBasePath = "product_listings"
+const productsListingResourceName = "product_listings"
+
+// ProductListingService is an interface for interfacing with the product listing endpoints
+// of the Shopify API.
+// See: https://shopify.dev/docs/admin-api/rest/reference/sales-channels/productlisting
+type ProductListingService interface {
+	List(interface{}) ([]ProductListing, error)
+	ListWithPagination(interface{}) ([]ProductListing, *Pagination, error)
+	Count(interface{}) (int, error)
+	Get(int64, interface{}) (*ProductListing, error)
+	GetProductIDs(interface{}) ([]int64, error)
+	Publish(int64) (*ProductListing, error)
+	Delete(int64) error
+}
```

Comment:
Should I change this to take the new ProductID type as a parameter rather than plain ints? Probably makes the code clearer. 

```
type ProductListingService interface {
	List(interface{}) ([]ProductListing, error)
	ListWithPagination(interface{}) ([]ProductListing, *Pagination, error)
	Count(interface{}) (int, error)
	Get(ProductID, interface{}) (*ProductListing, error)
	GetProductIDs(interface{}) ([]int64, error)
	Publish(ProductID) (*ProductListing, error)
	Delete(ProductID) error
}
```

Reply:
I would continue to use int64. 
Here are my reasons:
- it is consistent with other endpoints in the library
- it is less wordy when called. For example: `Delete(1234)` vs `Delete(NewProductID(1234))`